astropy/io/fits/hdu/base.py

```diff
@@ -457,7 +457,7 @@ def block_iter(nbytes):
         hdu._data_size = size + _pad_length(size)
 
         # Checksums are not checked on invalid HDU types
-        if checksum and checksum != 'remove' and isinstance(hdu, _ValidHDU):
+        if isinstance(hdu, _ValidHDU):
```

Comment:
@MSeifert04 Yes yes, I'll add a null check. Although all I want is `_verify_checksum_datasum` to be called even if checksum is False because that sets the _only_ function that sets `self._checksum` (set self._checksum to None) And if that's not set one will get `AttributeError` every time `-k none` option is use. Well may be I should modify that to make it little more logical!